app/utils/utils.py

```diff
@@ -0,0 +1,842 @@
+    pdf = EX()
+
+    pdf.set_title("OpenAgri Reporting Template")
+
+    pdf.add_page()
+
+    pdf.set_font("helvetica", "B", 10)
+    pdf.cell(40, 10, "Plant Protection")
+    pdf.set_font("helvetica", "", 8)
+
+    EX.ln(pdf)
+
+    keys = ["Date", "Enemy/Target", "Active Substance", "Product", "Dose", "Unit", "Area", "Treatment Description"]
+    with pdf.table() as table:
+        if len(ppp) != 0:
+            row = table.row()
+            for key in keys:
+                row.cell(key)
+            for p in ppp:
+                row = table.row()
+
+                if p.date is None:
+                    row.cell(p.date)
+                else:
+                    row.cell(str(p.date))
+
+                row.cell(p.enemy_target)
+                row.cell(p.active_substance)
+                row.cell(p.product)
+                row.cell(str(p.dose))
+                row.cell(p.unit)
+                row.cell(p.area)
+                row.cell(p.treatment_description)
+        else:
+            row = table.row()
+            for key in keys:
+                row.cell(key)
+            row = table.row()
+            for i in range(len(keys)):
+                row.cell(None)
+
+    return pdf
+
+
+def work_book(
+        farm: FarmProfile,
+        plot: List[PlotParcelDetail],
+        cult: List[GenericCultivationInformationForParcel],
+        irri: List[Irrigation],
+        fert: List[Fertilization],
+        pdmd: List[PestManagement]
+):
+
+    pdf = EX()
+
+    pdf.set_title("OpenAgri Reporting Template")
+
+    # 1
+
+    pdf.add_page()
+
+    pdf.set_font("helvetica", "B", 10)
+    pdf.cell(40, 10, "1. Farm profile")
+    pdf.set_font("helvetica", "", 8)
+
+    EX.ln(pdf)
+
+    keys = ["First name or surname", "Father's name", "VAT", "Head office details", "Phone", "District", "County", "Municipality", "Community", "Place name", "Farm area", "Plot IDs"]
+
+    with pdf.table() as table:
+        if farm is not None:
+            for key, value in zip(keys, list(farm.model_dump().values())):
+                row = table.row()
+                row.cell(key)
+                if value is not None:
+                    row.cell(str(value))
+                else:
+                    row.cell(value)
+        else:
+            for key in keys:
+                row = table.row()
+                row.cell(key)
+                row.cell(None)
+
+    EX.ln(pdf)
+
+    # 2
+
+    pdf.add_page()
+
+    pdf.set_font("helvetica", "B", 10)
+    pdf.cell(40, 10, "2. Machinery / Assets of Farm")
+    pdf.set_font("helvetica", "", 8)
+
+    EX.ln(pdf)
+
+    keys = ["Index", "Description", "Serial Number", "Date of manufacturing"]
+    with pdf.table() as table:
+        row = table.row()
+        for key in keys:
+            row.cell(key)
+        for i in range(1):
+            row = table.row()
+            for j in range(4):
+                row.cell(None)
+
+    # 3
+
+    first = 1
+    first_page = True
```

Comment:
They are used when adding plot information, because the plots are represented using images, and the images can be huge at points, I keep a count of what page (image) I'm on, so as to not cause either an image overlap, or have more than one image on a page.